cs/src/Connections/ITunnelClient.cs

```diff
@@ -26,6 +26,20 @@ public interface ITunnelClient : IAsyncDisposable
     /// </summary>
     IReadOnlyCollection<TunnelConnectionMode> ConnectionModes { get; }
 
+    /// <summary>
```

Comment:
What if `WaitForForwardedPortAsync` is called for the port that was cancelled?

Reply:
It will continue waiting until the port is forwarded (again) without being cancelled, which I think is correct behavior. If the client application both waits for and cancels the same port number, that is their bug.